### DISCUSSION TITLE
Tip For Creating Touring Routes

### DISCUSSION THREAD

**Discussion by Sonwon1** [osmandapp/OsmAnd#18103] [General]
This method will avoid the annoying micro instructions and the type of turns will all be correctly displayed that so often comes with imports of GPX files.

While the Snap To Road feature is a great tool I find it to slow to be useful on long routes, 200-300 kilometers.

For on road touring routes I discovered the easiest way for me is to map the route in Google Maps. Downside is Google has a limit of ten waypoints and you can no longer specify no dirt roads. The no dirt roads appears to be less of a problem since Google prefers to route onto the best quality road. And most routes up to 200-300 kilometers can be done in ten waypoints.

I first map the route in Google using extra waypoints as needed to force Google to route where I want. Then I manually transfer the route into OsmAnd by entering in the starting and ending waypoints. I then add the waypoints as Map Markers to force the routing where I want, not every waypoint needs to be added since OsmAnd often selects the route I want. Create all of the Map Markers before creating the Navigation route in OsmAnd.

Tip for finding the mid waypoints in OsmAnd. On Google Maps left clicking on the map will give you the GPS coordinates for the click point. Using OsmAnd search to find that GPS coordinate in OsmAnd and then adjust the point and save as a Map Marker.

After the route is created in OsmAnd I then save it as a GPX file. A OsmAnd GPX file will reload into OsmAnd correctly and not have the annoying micro instructions and the type of turns will all be correctly displayed. After I've saved the route I delete all of the Map Markers that were created.

I am not done yet, I then use Google Maps to find all of the spots along the route where I want to stop and create a Map Marker for each spot so I know when I arrive at that spot that I wanted to stop there. This avoids driving pass interesting sights along the road.

This method is not optimized so please do post any suggestions you may have to reduce the time it takes to create a road trip route.

This is the method I finally used after much trial and error to create a 16 day road tour. I will be testing the route next week.

Happy Tours!

**Comment by mikehgentry** [comment]
Try QMapShack / BRouter if you're on a computer rather than a phone. Bit of a pain to set up, but it makes better routes than Google I think (and is definitely vastly more customisable)

**Reply by Sonwon1** [reply]
Thank you, I will be checking it out when I return!

Google Maps actually works really well it is just under featured.

**Reply by scaidermern** [reply]
I also suggest to use [BRouter Web](https://brouter.de/brouter-web/) for route planning.

**Comment by DmitryAlexei** [comment]
Have you tried https://osmand.net/map ?
with pro subscription you can download tracks from browser directly to your device

**Reply by Sonwon1** [reply]
Not yet, I also need to be able to share the routes not sure if that is supported. Google Maps is easy for the person I share with. I need to check on OsmAnd/Map will also looked into that after this tour.

**Comment by graham8** [comment]
Sorry for the noob question.  How do you load a route from Google maps into OsmAnd?  It looks like you are having to enter every waypoint manually - can't it just be imported from kml/gpx?

**Comment by Sonwon1** [comment]
> Sorry for the noob question. How do you load a route from Google maps into OsmAnd? It looks like you are having to enter every waypoint manually - can't it just be imported from kml/gpx?

Read the original post. You can get the GPS coordinates from Google maps and then use search in osmand to find the GPS point and set a map market. Yes is manual, imports cause micro instructions problem.

**Reply by graham8** [reply]
> Read the original post.

Maybe you could read and answer my question "How do you load a route from Google maps into OsmAnd?"

What I'm expecting, coming new to this software, is to be able to import a set of waypoints (destinations) and then invoke the navigation function to plot the detailed route.  That doesn't seem possible.  It doesn't seem to have occured to the authors that users might want to do that.  Afaict you can only import 'tracks' which typically have every change in direction as a waypoint and that is why you are getting micro instructions.

**Reply by DmitryAlexei** [reply]
@graham8 if your .gpx track has waypoints, it can be imported as favorites and later used for navigation (standard navigation or navigation by markers)
Probably this article can be helpful https://osmand.net/docs/user/navigation/setup/markers-navigation#overview

**Reply by graham8** [reply]
Yes, I read that.  It's a very complicated procedure to do what other software does rather simply.  So it doesn't seem I've missed anything.  I hope in time you will be able to streamline the process.  Meanwhile, I'll stick to CoPilot and put up with the less detailed maps.  Thank you.

**Comment by Sonwon1** [comment]
> Yes, I read that. It's a very complicated procedure to do what other software does rather simply. So it doesn't seem I've missed anything. I hope in time you will be able to streamline the process. Meanwhile, I'll stick to CoPilot and put up with the less detailed maps. Thank you.

Google no longer supports importing and exporting gpx files. This is not an OsmAnd Problem. It is Google reducing the usefulness of their software.

Importing a list of waypoints is not supported at this time. It is a new feature request that is label as nice to have.

**Reply by graham8** [reply]
Yes, I understand now what OsmAnd and indeed BRouter mean by a route.  It's actually a track and doesn't identify/preserve the waypoints used to generate it or distinguish trackpoints that correspond to turning points.  Although BRouter appears to estimate which trackpoints are turning points by some algo.

"Nice to have".  Indeed :)
